tests/syntax/test_distributions.py

```diff
@@ -457,83 +471,140 @@ def test_namedtuple():
 ## Comparisons and control flow
 
 def test_comparison():
-    with pytest.raises(RuntimeParseError):
-        compileScenic('ego = Object with foo (Range(0, 1) > 0.5)')
+    with pytest.raises(RandomControlFlowError):
+        compileScenic('ego = new Object with foo (Range(0, 1) > 0.5)')
 
 def test_len():
     # Taking the len of a distribution used to be illegal, but is now allowed
-    scenario = compileScenic('ego = Object with foo len(Uniform([0], [1, 2]))')
+    scenario = compileScenic('ego = new Object with foo len(Uniform([0], [1, 2]))')
     fs = [sampleEgo(scenario).foo for i in range(60)]
     assert all(f == 1 or f == 2 for f in fs)
     assert any(f == 1 for f in fs)
     assert any(f == 2 for f in fs)
 
 def test_iter():
-    with pytest.raises(RuntimeParseError):
+    with pytest.raises(RandomControlFlowError):
         compileScenic("""
             for x in Uniform([1, 2], [3, 4]):
-                ego = Object at x@0
+                ego = new Object at x@0
         """)
 
 def test_control_flow():
-    with pytest.raises(RuntimeParseError):
+    with pytest.raises(RandomControlFlowError):
         compileScenic("""
             if Uniform(False, True):
-                ego = Object
+                ego = new Object
             else:
-                ego = Object at 1@1
+                ego = new Object at 1@1
         """)
 
 ## Reproducibility
+@pytest.mark.slow
+def test_specifier_order():
+    values = []
+    
+    paramPattern = re.compile(r'\s*Parameter "p": (.*)$')
+    seed = random.randint(0,10000)
+
+    for _ in range(100):
```

Comment:
This test takes almost a minute. If the specifier order becomes random a single iteration should have a ~50% chance of failure, so let's reduce the number of iterations to 30 (so the false negative rate should be about 1 in 10^9).